compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -1614,6 +1614,26 @@ object SymDenotations {
 
       annotations.collect { case Annotation.Child(child) => child }.reverse
     end children
+
+    /** Recursively assemble all children of this symbol, including this symbol.
+     *  Preserves order of insertion.
+     */
+    final def sealedDescendants(using Context): List[Symbol] =
+      @tailrec
+      def loop(syms: List[Symbol], acc: List[Symbol]): List[Symbol] = syms match
+        case sym :: syms1 =>
+          val notSeen = sym.children.filterConserve(!acc.contains(_))
+          if notSeen.isEmpty then
+            loop(syms1, acc)
+          else
+            loop(syms1 ::: notSeen, acc ::: notSeen)
```

Comment:
Appending into a ListBuffer would be more efficient

Reply:
I have refactored to optimise for the case of a single level deep hierarchy